examples/hyperparam/search_gpyopt.py

```diff
@@ -89,33 +89,32 @@ def eval(params):
             :return: The metric value evaluated on the validation data.
             """
             lr, momentum = params[0]
-            p = mlflow.projects.run(
-                uri=".",
-                entry_point="train",
-                parameters={
-                    "training_data": training_data,
-                    "epochs": str(nepochs),
-                    "learning_rate": str(lr),
-                    "momentum": str(momentum),
-                    "seed": str(seed)},
-                experiment_id=experiment_id,
-                synchronous=False
-            )
-
-            if p.wait():
+            with mlflow.start_run(nested=True) as child_run:
```

Comment:
Hm why is this necessary? It doesn't look like `projects.run` mutates the current active run (i.e.  it calls `create_run` instead of `start_run`, so we should be able to just call `projects.run` I think? Unless there's something I'm missing

Reply:
This is necessary so to keep have the parent run id set in the child run. process.run does not do it (I tired to pass in the parent run id but that does not work at all). The alternative would be to use the client api to set the tag manually, but I think this one is cleaner, especially if we would ever decide to inherit more tags.